src/streamlink/plugins/zdf_mediathek.py

```diff
@@ -29,9 +30,12 @@
     )
 }
 
-_url_re = re.compile(r"""
+_url_zdf_re = re.compile(r"""
     http(s)?://(\w+\.)?zdf.de/
 """, re.VERBOSE | re.IGNORECASE)
+_url_3sat_re = re.compile(r"""
+    http(s)?://(\w+\.)?3sat.de/
+""", re.VERBOSE | re.IGNORECASE)
```

Comment:
Please only use one regex with a capture group for the domain names.

Then select the right API URL from the capture group value, or build the API URL dynamically.

Reply:
Done. I'm now extracting the API base URL from the 1st response and dropped the hardcoded API URL(s). Note that I'm neither very familiar with streamlink internals nor with web APIs in general, so I hope this makes sense. At least it worked fine for 2 test URLs I tried.